src/scripts/main.js

```diff
@@ -1,3 +1,38 @@
 'use strict';
 
-// write your code here
+const logo = document.querySelector('.logo');
+
+function createPromise(message) {
+  return new Promise((resolve, reject) => {
+    if (!message) {
+      resolve('Promise was resolved!');
+    } else {
+      setTimeout(() => reject(message), 3000);
+    }
+  });
+}
+
+function handleLogoClick() {
+  const messageDiv = document.createElement('div');
+
+  messageDiv.classList.add('message');
+
+  const promise1 = createPromise();
+  const promise2 = createPromise('Promise was rejected!');
+
+  promise1.then((result) => {
+    messageDiv.innerText = result;
+    document.body.appendChild(messageDiv);
+  });
+
+  setTimeout(() => {
+    promise2.catch((error) => {
+      messageDiv.classList.add('error-message');
+      messageDiv.innerText = error;
+    });
+  }, 3000);
```

Comment:
Here, you are catching the rejection of `promise2` after a delay of 3 seconds. However, `promise2` is already set to reject after 3 seconds in line 21. Therefore, this `setTimeout` function is unnecessary. You can catch the rejection immediately after it occurs by chaining a `.catch` block to the promise right after it is created.